apps/protocol/src/app/hooks/useMassetExchangeRate.ts

```diff
@@ -52,45 +52,38 @@ export const useExchangeRateForMassetInputs = (
 
 export const useExchangeRateForFPInputs = (
   poolAddress: string,
-  estimatedOutputAmount?: FetchState<BigDecimal>,
+  estimatedOutputAmount: FetchState<BigDecimal>,
   inputValues?: BigDecimalInputValues,
 ): FetchState<BigDecimal> => {
   const feederPool = useSelectedFeederPoolState()
 
   return useMemo(() => {
     if (!inputValues) return {}
 
-    const touched = Object.values(inputValues).filter(v => v.touched)
+    const touched = Object.values(inputValues).filter(v => v.touched && v.amount.exact.gt(0))
 
     if (!touched.length) return {}
-    if (estimatedOutputAmount?.error) return {}
-    if (!estimatedOutputAmount?.value) return { fetching: true }
-
-    // Scale asset via ratio
-    const scaleAssetValue = (input: BigDecimalInputValue): BigDecimal => {
-      const { amount } = input
-      if (!amount) return BigDecimal.ZERO
-
-      if (input.address === feederPool.masset.address) {
-        // The mAsset amount is not scaled
-        return amount
-      }
+    if (estimatedOutputAmount.error) return {}
+    if (!estimatedOutputAmount.value) return { fetching: true }
 
-      // The fAsset amount can be scaled
-      return amount.mulRatioTruncate(feederPool.fasset.ratio)
-    }
+    const inputAmount = Object.values(touched).reduce<BigDecimal>((prev, { address, amount }) => {
+      if (address === feederPool.masset.address) return prev.add(amount)
 
-    const totalAmount = Object.values(touched).reduce((prev, v) => prev.add(scaleAssetValue(v)), BigDecimal.ZERO)
+      let scaledAmount = amount.mulRatioTruncate(feederPool.fasset.ratio).setDecimals(18)
 
-    if (totalAmount) {
-      if (estimatedOutputAmount.value.exact.eq(0) || totalAmount.exact.eq(0)) {
-        return { error: 'Output amount must be greater than zero' }
+      if (feederPool.fasset.token.symbol === 'RAI') {
+        // For RAI, scale back down; the ratio is the relative price :)
+        scaledAmount = feederPool.fasset.price.divPrecisely(scaledAmount)
```

Comment:
The ratio is `302774230` to scale it to masset size. For assets with different decimals, we still need the first step, but here, we need to undo that (retaining the correct units)